File: classic_gym/classic_gym/report/calories_intake_report/calories_intake_report.js
// Copyright (c) 2024, Polito and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Calories Intake Report"] = {
	"filters": [
		{
		"fieldname" : "member_email",
		"label":__('Member Email'),
		"fieldtype": 'Link',
		"options":"Gym Member",
		 "width": 200,
		"reqd": 0

		},
		{
				
			"fieldname": "age",
            "label": __("Age"),
            "fieldtype": "Data",
            "width": 200,
            "reqd": 0
		
				},
		{
			"fieldname" : "calorie_intake",
			"label":__('Calorie Intake'),
			"fieldtype": 'Link',
			"options":"Calorie Intake",
			"width": 200,
			"reqd": 0
	
			},
		{
				
			"fieldname": "date",
            "label": __("Date"),
            "fieldtype": "Date",
            "width": 200,
            "reqd": 0
		
				}
	]
};

File: classic_gym/classic_gym/report/calories_intake_report/calories_intake_report.py
# Copyright (c) 2024, Polito and contributors
# For license information, please see license.txt

import frappe

def execute(filters=None):
    columns = get_columns()
    data = get_data(filters)
    return columns, data


def get_data(filters):
    conditions = "1=1"
    if filters.get("member_email"):
        conditions += f" AND gm.email='{filters.get('member_email')}'"

    SQL = f"""
        SELECT 
            gm.email,
            gm.age,
            ci.calorie_intake,
            ci.date
        FROM
            `tabGym Member` AS gm
        JOIN 
            `tabCalorie Intake` AS ci
        ON
            gm.email = ci.parent
        WHERE
            {conditions}
    """
    data = frappe.db.sql(SQL)
    return data


def get_columns():
    return [
        "Member Name:Link/Gym Member:200",
        "Age:Data:200",
        "Calorie Intake:Link/Calorie Intake:250",
        "Date:Date:200"
        
    ]

